### ISSUE TITLE
ci-change: git checkout behaviour of `node-test-commit-v8-linux`

### ISSUE THREAD

**Issue by refack** [nodejs/build#1576]
Due to a suspected bug leading to lingering file conflicts (https://github.com/nodejs/node/issues/24393#issuecomment-439632502), I've tweaked the git parameters of [`node-test-commit-v8-linux`](https://ci.nodejs.org/job/node-test-commit-v8-linux/jobConfigHistory/showDiffFiles?timestamp1=2018-11-13_14-21-36&timestamp2=2018-11-18_09-31-34)

Example run with `master`: https://ci.nodejs.org/job/node-test-commit-v8-linux/1880/
Example run with `node-v8/canary`: https://ci.nodejs.org/job/node-test-commit-v8-linux/1880/
Example run with an arbitrary git SHA: https://ci.nodejs.org/job/node-test-commit-v8-linux/1869/
Example run with a branch off `v8.x`: https://ci.nodejs.org/job/node-test-commit-v8-linux/1882/
(All test seem to checkout code correctly. Failures are code based and are reported bugs).
/CC @targos @bethgriggs

**Comment by mhdawson** [comment]
Would it be possible to summarize in this issue what the tweaks were or at links to where to see them?

**Comment by refack** [comment]
The [`node-test-commit-v8-linux`](https://ci.nodejs.org/job/node-test-commit-v8-linux/jobConfigHistory/showDiffFiles?timestamp1=2018-11-13_14-21-36&timestamp2=2018-11-18_09-31-34) link is to the diff of the changes.

tl;dr
1. Added `clean before checkout`
2. Added `checkout local branch = _jenkins_local_branch` to the git plugin instead of it being called from the bash script.
3. Added shallow cloning, depth=100 (which is only relevant after workspace is cleared)
4. made `git rebase --abort` conditional on actually doing a rebase
5. Removed unused IGNORE_FLAKY_TESTS since it's hardcoded to `dontcare` in the job script
6. removed duplicate `git config user.name`

**Comment by refack** [comment]
Done.
